### PR TITLE
tests/coredump-options: test core dump generation

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
The needed changes in the base are now in edge and beta core24 channels: make sure we are generating the core dumps.

### CODE REVIEW

File: tests/core/coredump-options/task.yaml [hunk 1]
```diff
@@ -6,16 +6,33 @@ details: |
...
   expect=$(printf "[Coredump]\nStorage=none\nProcessSizeMax=0\n")
   test "$expect" = "$(cat $cfg_path)"
 
-  # TODO no coredumps are dropped, but we get a journal entry. Depends
-  # on https://github.com/snapcore/core-base/pull/227 being in the
-  # installed core24 channel already.
+  # Install snap with a service that simply waits
+  core_dump_d=$TESTSLIB/snaps/core-dump-snap
+  BASE=core"$VERSION_ID"
+  sed -i "s/##BASE##/$BASE/" "$core_dump_d"/meta/snap.yaml
```
Comment (sergiocazzolato) [inline tests/core/coredump-options/task.yaml:25]: we should revert this change during restore

File: tests/core/coredump-options/task.yaml [hunk 2]
```diff
@@ -6,16 +6,33 @@ details: |
...
+  DUMP_SNAP_FILE=core-dump-snap.snap
+  snap pack "$core_dump_d" --filename="$DUMP_SNAP_FILE"
+  snap install --dangerous "$DUMP_SNAP_FILE"
+
+  # try to generate dump
+  CRASH_PID=$(retry --wait 1 -n 5 pgrep crash.sh)
+  kill -ABRT "$CRASH_PID"
+  retry --wait 1 -n 5 sh -c "coredumpctl | MATCH 'none */bin/bash'"
+  # This cleans-up the coredumctl list
+  sudo journalctl --rotate; sudo journalctl --vacuum-time=1ms
```
Comment (sergiocazzolato) [inline tests/core/coredump-options/task.yaml:35]: sudo is not needed

File: tests/core/coredump-options/task.yaml [hunk 3]
```diff
@@ -25,13 +42,17 @@ execute: |
...
-  # TODO Generate a core dump. Depends on
-  # https://github.com/snapcore/core-base/pull/227 too.
+  # Force generation of core dump and check it is present
+  CRASH_PID=$(retry --wait 1 -n 5 pgrep crash.sh)
+  kill -ABRT "$CRASH_PID"
+  retry --wait 1 -n 5 sh -c "coredumpctl | MATCH 'none */bin/bash'"
+  coredumpctl dump --output=core
+  test -s core
+  # This cleans-up the coredumctl list and the core dump
+  sudo journalctl --rotate; sudo journalctl --vacuum-time=1ms
```
Comment (sergiocazzolato) [inline tests/core/coredump-options/task.yaml:52]: same about sudo

File: tests/core/core-dump/task.yaml [hunk 4]
```diff
@@ -7,16 +7,17 @@ details: |
...
   # To get VERSION_ID defined
   . /etc/os-release
+  core_dump_d=$TESTSLIB/snaps/core-dump-snap
   if [ "$VERSION_ID" = 16 ]; then
-      sed -i '/base: ##BASE##/d' core-dump-snap/meta/snap.yaml
+      sed -i '/base: ##BASE##/d' "$core_dump_d"/meta/snap.yaml
   else
       BASE=core"$VERSION_ID"
-      sed -i "s/##BASE##/$BASE/" core-dump-snap/meta/snap.yaml
+      sed -i "s/##BASE##/$BASE/" "$core_dump_d"/meta/snap.yaml
```
Comment (sergiocazzolato) [inline tests/core/core-dump/task.yaml:15]: is has to be reverted during restore

**Review by sergiocazzolato — COMMENTED** [review]
Minor comments inline

**Review by Meulengracht — APPROVED** [review]
+1 but agree with sergio's comments

File: tests/core/core-dump/task.yaml (hunk 4 above)
Comment (alfonsosanchezbeato) [reply tests/core/core-dump/task.yaml:16]: Good point, what I've done now is to copy inside the test folder before modifying.

File: tests/core/coredump-options/task.yaml (hunk 1 above)
Comment (alfonsosanchezbeato) [reply tests/core/coredump-options/task.yaml:26]: Changed to copying inside the test folder before modifying.

**Review by alfonsosanchezbeato — COMMENTED** [review]
Thanks for the reviews, I've addressed the comments now

File: tests/core/coredump-options/task.yaml (hunk 3 above)
Comment (alfonsosanchezbeato) [reply tests/core/coredump-options/task.yaml]: Removed

**Review by sergiocazzolato — APPROVED** [review]
Thanks
